manim/mobject/types/vectorized_mobject.py

```diff
@@ -2107,30 +2107,22 @@ def __init__(
         self.num_dashes = num_dashes
         self.positive_space_ratio = positive_space_ratio
         super().__init__(color=color, **kwargs)
-        ps_ratio = self.positive_space_ratio
+        p = self.positive_space_ratio
+        n = self.num_dashes
         if num_dashes > 0:
-            # End points of the unit interval for division
-            alphas = np.linspace(0, 1, num_dashes + 1)
-
-            # This determines the length of each "dash"
-            full_d_alpha = 1.0 / num_dashes
-            partial_d_alpha = full_d_alpha * ps_ratio
-
-            # Shifts the alphas and removes the last dash
-            # to give closed shapes even spacing
+            # Assuming total length is 1
```

Comment:
I don't understand this, what is the total length ?

Reply:
total length is like the vmobject's perimeter which we assume as 1 map it from 0% to 100% of the vmobject's subcurve... same as alpha interpolating from 0 to 1 as it was before the changes... but it was really complicated to understand and fix for me

Reply:
I get the idea, but I don't understand how it's linked to this line. It's just the space ratio divided by the number of dashes, no ?

Reply:
Yes, we are basically dividing our unit length into p : (1-p), and then splitting it into n parts, hence p/n and (1-p)/n are dash_len and void_len respectively